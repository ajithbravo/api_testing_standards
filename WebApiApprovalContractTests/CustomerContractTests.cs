using NUnit.Framework;
using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.Owin.Testing;
using WebApiTestingStandards;

namespace WebApiApprovalContractTests
{
    [UseReporter(typeof(DiffReporter))]
    public class CustomerContractTests
    {
        [Test]
        public void CustomerContract_VerifyTheCustomerContract_ApproveTheCurrentVersion()
        {
            using (var server = TestServer.Create<Startup>())
            {
                server.BaseAddress = new Uri("http://localhost:4000/swagger/docs/v1");
                var response = server.HttpClient.GetAsync(server.BaseAddress)
                    .Result.Content.ReadAsStringAsync();
                Approvals.VerifyJson(response.Result);
            }
        }
    }
}