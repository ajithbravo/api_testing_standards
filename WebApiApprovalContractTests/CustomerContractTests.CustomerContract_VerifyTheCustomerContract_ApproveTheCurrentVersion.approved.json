{
    "swagger": "2.0",
    "info": {
        "version": "v1",
        "title": "CustomersApi"
    },
    "host": "localhost:4000",
    "schemes": [
        "http"
    ],
    "paths": {
        "/api/customers": {
            "get": {
                "tags": [
                    "Customer"
                ],
                "operationId": "Customer_Get",
                "consumes": [],
                "produces": [
                    "application/json",
                    "text/json",
                    "text/html",
                    "application/xml",
                    "text/xml"
                ],
                "responses": {
                    "200": {
                        "description": "OK",
                        "schema": {
                            "$ref": "#/definitions/CustomerSearchResults"
                        }
                    }
                }
            }
        },
        "/api/customer": {
            "get": {
                "tags": [
                    "Customer"
                ],
                "operationId": "Customer_GetCustomer",
                "consumes": [],
                "produces": [
                    "application/json",
                    "text/json",
                    "text/html",
                    "application/xml",
                    "text/xml"
                ],
                "parameters": [
                    {
                        "name": "id",
                        "in": "query",
                        "required": true,
                        "type": "integer",
                        "format": "int32"
                    }
                ],
                "responses": {
                    "200": {
                        "description": "OK",
                        "schema": {
                            "type": "object"
                        }
                    }
                }
            }
        },
        "/api/newCustomer": {
            "post": {
                "tags": [
                    "Customer"
                ],
                "operationId": "Customer_Post",
                "consumes": [
                    "application/json",
                    "text/json",
                    "text/html",
                    "application/xml",
                    "text/xml",
                    "application/x-www-form-urlencoded"
                ],
                "produces": [
                    "application/json",
                    "text/json",
                    "text/html",
                    "application/xml",
                    "text/xml"
                ],
                "parameters": [
                    {
                        "name": "request",
                        "in": "body",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/NewCustomerRequest"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "OK",
                        "schema": {
                            "$ref": "#/definitions/Customer"
                        }
                    }
                }
            }
        }
    },
    "definitions": {
        "CustomerSearchResults": {
            "type": "object",
            "properties": {
                "Customers": {
                    "type": "array",
                    "items": {
                        "$ref": "#/definitions/Customer"
                    }
                }
            }
        },
        "Customer": {
            "type": "object",
            "properties": {
                "Name": {
                    "type": "string"
                },
                "Id": {
                    "format": "int32",
                    "type": "integer"
                },
                "Address": {
                    "type": "string"
                },
                "Telephone": {
                    "type": "string"
                },
                "Salary": {
                    "format": "double",
                    "type": "number"
                }
            }
        },
        "NewCustomerRequest": {
            "required": [
                "Name",
                "Address"
            ],
            "type": "object",
            "properties": {
                "Name": {
                    "type": "string"
                },
                "Address": {
                    "type": "string"
                },
                "Telephone": {
                    "type": "string"
                }
            }
        }
    }
}